pramen/extras/src/main/scala/za/co/absa/pramen/extras/sink/EnceladusUtils.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.pramen.extras.sink
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.encoders.ExpressionEncoder
+import org.apache.spark.sql.functions.col
+import org.slf4j.LoggerFactory
+import za.co.absa.pramen.extras.query.QueryExecutor
+import za.co.absa.pramen.extras.utils.FsUtils
+import za.co.absa.pramen.extras.utils.PartitionUtils.unpackCustomPartitionPattern
+
+import java.time.LocalDate
+import scala.util.matching.Regex
+import scala.util.{Failure, Success, Try}
+
+class EnceladusUtils(rawPartitionPattern: String,
+                     publishPartitionPattern: String,
+                     infoDateColumn: String)
+                    (implicit spark: SparkSession,
+                     queryExecutor: QueryExecutor) {
+  private val log = LoggerFactory.getLogger(this.getClass)
+
+  val rawVersionRegEx: Regex = "^.*v(\\d+)$".r
+  val publishVersionRegEx: Regex = "^.*=(\\d+)$".r
+
+  def getNextEnceladusVersion(infoDate: LocalDate,
```

Comment:
I've thought of using Either initially, but then realized that it would look like (in pseudocode) 😆  :
```
Either[Option[Int],Failure[Throwable]]
```

I like your suggestion since it expresses the intent better. Will play with it, maybe I can get rid of `Try[Option[Int]]`

Reply:
The new code with the case classes looks great!

Reply:
Thanks!